chain/epoch_manager/src/reward_calculator.rs

```diff
@@ -83,47 +88,127 @@ impl RewardCalculator {
         }
         let epoch_validator_reward = epoch_total_reward - epoch_protocol_treasury;
         let mut epoch_actual_reward = epoch_protocol_treasury;
-        let total_stake: Balance = validator_stake.values().sum();
-        for (account_id, stats) in validator_block_chunk_stats {
-            // Uptime is an average of block produced / expected and chunk produced / expected.
-            let average_produced_numer = U256::from(
-                stats.block_stats.produced * stats.chunk_stats.expected
-                    + stats.chunk_stats.produced * stats.block_stats.expected,
-            );
-            let average_produced_denom =
-                U256::from(2 * stats.chunk_stats.expected * stats.block_stats.expected);
-            let online_min_numer = U256::from(*self.online_min_threshold.numer() as u64);
-            let online_min_denom = U256::from(*self.online_min_threshold.denom() as u64);
-            // If average of produced blocks below online min threshold, validator gets 0 reward.
-            let reward = if average_produced_numer * online_min_denom
-                < online_min_numer * average_produced_denom
-                || stats.chunk_stats.expected == 0
-                || stats.block_stats.expected == 0
-            {
-                0
-            } else {
+
+        if checked_feature!(
+            "protocol_feature_chunk_only_producers",
+            ChunkOnlyProducers,
+            protocol_version
+        ) {
+            let to_big_rational = |rational: Rational| -> BigRational {
+                BigRational::new((*rational.numer()).into(), (*rational.denom()).into())
+            };
+            let online_min_threshold = to_big_rational(self.online_min_threshold);
+            let online_max_threshold = to_big_rational(self.online_max_threshold);
+            let calc_uptime_mult = |stats: &ValidatorStats| -> BigRational {
```

Comment:
add a comment that this is the function, that we'll later apply to both block and chunk producers.